### PR TITLE
Make genezio deploy, account and list automatically login if token is incorrect

### PR DESCRIPTION
**Pull request by costinsin**
## Type of change

-   [x] 🧑‍💻 Improvement

## Description
This makes the user flow easier because we login automatically for him if his token is missing or incorrect.

## Checklist

-   [x] My code follows the contributor guidelines of this project;
-   [x] New and existing unit tests pass locally with my changes;

### CODE REVIEW

**Comment by costinsin** [comment]
Current dependencies on/for this PR:
* `dev`
  * **PR #699** <a href="https://app.graphite.dev/github/pr/Genez-io/genezio/699?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/Genez-io/genezio/699?utm_source=stack-comment).

File: src/commands/superGenezio/superGenezio.ts [hunk 1]
```diff
@@ -48,11 +45,6 @@ export async function genezioCommand() {
                     installDeps: true,
                 };
 
-                if (!(await getAuthToken())) {
```
Comment (radu1122) [inline src/commands/superGenezio/superGenezio.ts:51]: why did you remove this?

File: src/utils/accounts.ts [hunk 2]
```diff
@@ -18,6 +19,15 @@ export async function getAuthToken(): Promise<string | undefined> {
     }
 }
 
+export async function isLoggedIn(): Promise<boolean> {
+    try {
+        await getUser();
```
Comment (radu1122) [inline src/utils/accounts.ts:24]: is this working? `getUser()` will throw err if there is no user connected?

File: src/commands/superGenezio/superGenezio.ts (hunk 1 above)
Comment (costinsin) [reply src/commands/superGenezio/superGenezio.ts:51]: Because the auto login is done now in the `deployCommand` function. We no longer need to auto login here.

Comment (radu1122) [reply src/commands/superGenezio/superGenezio.ts:51]: let s also keep it here

File: src/utils/accounts.ts (hunk 2 above)
Comment (costinsin) [reply src/utils/accounts.ts:24]: Yes, it will throw `Error(GENEZIO_NOT_AUTH_ERROR_MSG)` if no token was found

Comment (radu1122) [reply src/utils/accounts.ts:24]: yes, but there is also the case when there is a token. but it is invalid

Comment (costinsin) [reply src/utils/accounts.ts:24]: I tested it, stop making guesses please

**Review by radu1122 — APPROVED** [review]
